src/cdk/tree/tree.ts

```diff
@@ -263,22 +263,22 @@ export class CdkTree<T> implements CollectionViewer, OnInit, OnDestroy {
 
     if (dataStream) {
       this._dataSubscription = dataStream.pipe(takeUntil(this._onDestroy))
-        .subscribe(data => this._renderNodeChanges(data));
+        .subscribe(data => this.renderNodeChanges(data));
     } else {
       throw getTreeNoValidDataSourceError();
     }
   }
 
   /** Check for changes made in the data and render each change (node added/removed/moved). */
-  private _renderNodeChanges(dataNodes: T[]) {
-    const changes = this._dataDiffer.diff(dataNodes);
+  renderNodeChanges(data: T[], dataDiffer: IterableDiffer<T> = this._dataDiffer,
+                    viewContainer: ViewContainerRef = this._nodeOutlet.viewContainer) {
+    const changes = (dataDiffer ? dataDiffer : this._dataDiffer).diff(data);
```

Comment:
Now that you're guaranteed to have a `dataDiffer`, you don't need the ternary here anymore.